x/mint/keeper/keeper.go

```diff
@@ -279,15 +300,23 @@ func (k Keeper) distributeDeveloperRewards(ctx sdk.Context, totalMintedCoin sdk.
 	return devRewardCoin.Amount, nil
 }
 
-// getProportions gets the balance of the `MintedDenom` from minted coins and returns coins according to the
-// allocation ratio. Returns error if ratio is greater than 1.
-// TODO: this currently rounds down and is the cause of rounding discrepancies.
-// To be fixed in: https://github.com/osmosis-labs/osmosis/issues/1917
-func getProportions(mintedCoin sdk.Coin, ratio sdk.Dec) (sdk.Coin, error) {
-	if ratio.GT(sdk.OneDec()) {
-		return sdk.Coin{}, invalidRatioError{ratio}
+// calculateTotalTruncationDelta returns the total truncation delta that has not been distributed yet
+// for the given module account, provisions, and actual amount distributed in the current epoch.
+// The returned delta might include the delta from previous epochs if they have not been distributed due to truncations.
+// For example, assume that for some number of epochs our expected provisions
+// are 100.6 and the actual amount distributed is 100 every epoch due to truncations.
+// Then, at epoch 1, we have a delta of 0.6. 0.6 cannot be distributed because it is not an integer.
+// So we persist it until the next epoch. Then, at at epoch 2, we get a delta of 1.2 (0.6 from epoch 1 and 0.6 from epoch 2).
+// Now, 1 can be distributed and 0.2 gets persisted until the next epoch.
+// CONTRACT: provisions are greater than or equal to amountDistributed.
```

Comment:
What does contract here mean?

Reply:
it pretty much means an invariant that must hold